steps/transform.py

```diff
@@ -1,64 +1,19 @@
-"""
-This module defines the following routines used by the 'transform' step of the regression pipeline:
-
-- ``transformer_fn``: Defines customizable logic for transforming input data before it is passed
-  to the estimator during model inference.
-"""
-
 from pandas import DataFrame
-from sklearn.compose import ColumnTransformer
 from sklearn.pipeline import Pipeline
-from sklearn.preprocessing import OneHotEncoder, StandardScaler, FunctionTransformer
+from sklearn.preprocessing import FunctionTransformer
 
 
 def calculate_features(df: DataFrame):
-    """
-    Extend the input dataframe with pickup day of week and hour, and trip duration.
-    Drop the now-unneeded pickup datetime and dropoff datetime columns.
-    """
-    df["pickup_dow"] = df["tpep_pickup_datetime"].dt.dayofweek
-    df["pickup_hour"] = df["tpep_pickup_datetime"].dt.hour
-    trip_duration = (
-            df["tpep_dropoff_datetime"] - df["tpep_pickup_datetime"]
-    )
-    df["trip_duration"] = trip_duration.map(lambda x: x.total_seconds() / 60)
```

Comment:
The quality of the model produced in the example will suffer if we don't do this cleaning. I think it's best to preserve this.